src/stellar-plus/horizon/index.ts

```diff
@@ -17,7 +17,14 @@ export class HorizonHandlerClient implements HorizonHandler {
    */
   constructor(networkConfig: NetworkConfig) {
     this.networkConfig = networkConfig
-    this.server = new Horizon.Server(this.networkConfig.horizonUrl)
+
+    if (!this.networkConfig.horizonUrl) {
+      throw DHHError.missingHorizonUrl()
+    }
+
+    const serverOpts = networkConfig.allowHttp ? { allowHttp: true } : {}
```

Comment:
Here we could simply set the value. WDYT?
```suggestion
    const serverOpts = { allowHttp: networkConfig.allowHttp }
```